hvplot/converter.py

```diff
@@ -1485,7 +1485,7 @@ def _process_style(self, kwds, plot_opts):
         if 'color' in style_opts:
             color = style_opts['color']
         elif not isinstance(cmap, dict):
-            if cmap and any(c in cmap for c in categories):
+            if isinstance(cmap, str) and any(c in cmap for c in categories):
```

Comment:
I have a vague feeling that this check is for `cmap=["red", "green", "blue"]` to make `any(c in cmap for c in categories)` make sense so I think the correct check would be to `if not isinstance(cmap, LinearSegmentedColormap)` or checking if it's iterable but I could be wrong.

Reply:
I'm not sure what you mean but I tried with a list of named colors and one of the items in the `categories` as well:

![image](https://github.com/user-attachments/assets/12ff5f81-50ee-403f-809b-9f4c7b8c27d5)

![image](https://github.com/user-attachments/assets/617340ef-307c-4bb2-885f-612951ca1573)

Reply:
Oh I misunderstood. `categories` is a `cmap` e.g. it's `categories_cmap` and not a list of hexcodes.

In that case, I think you could check if LinearSegmentedColormap before and if it is, then convert the linear segmented colormap into a string `get_cmap("RdBu_r").name`

Reply:
> In that case, I think you could check if LinearSegmentedColormap before and if it is, then convert the linear segmented colormap into a string get_cmap("RdBu_r").name

Do you mean something like this?

```python
elif not isinstance(cmap, dict):
    if isinstance(cmap, str) and any(c in cmap for c in categories):
        color = process_cmap(cmap or self._default_cmaps['categorical'], categorical=True)
    elif isinstance(cmap, ListedColormap):
        cmap = # do conversion here
        color = cmap
    else:
        color = cmap
else:
    color = style_opts.get('color')
```

Reply:
Something like this perhaps:

```
    if (isinstance(cmap, str) and any(c in cmap for c in categories)) or (isinstance(cmap, LinearSegmentedColormap) and any(c in cmap.name for c in categories)):
```